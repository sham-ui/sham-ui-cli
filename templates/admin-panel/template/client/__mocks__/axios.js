const METHODS = [
    'get',
    'post',
    'put',
    'delete'
];

const FIXTURES = {};
const MOCKS = {};
const INTERCEPTORS = {};

function mockMethodFactory( method ) {
    return jest.fn().mockImplementation(
        ( url ) => {
            if ( !FIXTURES[ method ].hasOwnProperty( url ) ) {
                throw new Error( `Missing fixture for ${method.toUpperCase()} url = "${url}"` );
            }
            const request = {
                url,
                method,
                headers: {}
            };
            INTERCEPTORS.request( request );
            const { status, data, headers } = FIXTURES[ method ][ url ];
            if ( 200 === status ) {
                const response = {
                    data,
                    headers,
                    status
                };
                return Promise.resolve(
                    INTERCEPTORS.response.success( response )
                );
            } else {
                const error = {
                    config: {
                        url,
                        method,
                        baseURL: '/'
                    },
                    response: {
                        data,
                        headers,
                        status
                    }
                };
                return INTERCEPTORS.response.fail( error );
            }
        }
    );
}

METHODS.forEach(
    method => {
        MOCKS[ method ] = mockMethodFactory( method );
        FIXTURES[ method ] = {};
    }
);

const defaultMocksData = {
    csrfToken: 'csrf-token',
    user: {
        Name: 'John Smith',
        Email: 'j.smith@example.com'
    }
};

export default {
    defaultMocksData: defaultMocksData,
    mocks: MOCKS,
    create: jest.fn().mockImplementation( () => {
        const instance  = {
            ...MOCKS,
            request( { url, method = 'get', data } ) {
                const args = [ url ];
                if ( undefined !== data ) {
                    args.push( data );
                }
                return instance[ method ].apply( null, args );
            },
            interceptors: {
                request: {
                    use( interceptor ) {
                        INTERCEPTORS.request = interceptor;
                    }
                },
                response: {
                    use( success, fail ) {
                        INTERCEPTORS.response = {
                            success,
                            fail
                        };
                    }
                }
            }
        };
        return instance;
    } ),
    use( method, url, data, status = 200, headers = {} ) {
        if ( !METHODS.includes( method ) ) {
            throw new Error( `Unknown method: "${method}"` );
        }
        FIXTURES[ method ][ url ] = {
            data,
            status,
            headers
        };
        return this;
    },

    useDefaultMocks() {
        this
            .use( 'get', '/csrftoken', '', 200, { 'x-csrf-token': defaultMocksData.csrfToken } )
            .use( 'get', '/validsession', defaultMocksData.user );
        return this;
    }
};
