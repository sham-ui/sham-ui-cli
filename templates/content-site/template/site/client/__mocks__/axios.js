const METHODS = [
    'get',
    'post',
    'put',
    'delete'
];

const FIXTURES = {};
const MOCKS = {};
const INTERCEPTORS = {};

function mockMethodFactory( method ) {
    return jest.fn().mockImplementation(
        ( url ) => {
            if ( !FIXTURES[ method ].hasOwnProperty( url ) ) {
                throw new Error( `Missing fixture for ${method.toUpperCase()} url = "${url}"` );
            }
            const request = {
                url,
                method,
                headers: {}
            };
            if ( INTERCEPTORS.request ) {
                INTERCEPTORS.request( request );
            }
            const { status, data, headers } = FIXTURES[ method ][ url ];
            if ( 200 === status ) {
                const response = {
                    data,
                    headers,
                    status
                };
                return Promise.resolve(
                    INTERCEPTORS.response.success( response )
                );
            } else {
                const error = {
                    config: {
                        url,
                        method,
                        baseURL: '/'
                    },
                    response: {
                        data,
                        headers,
                        status
                    }
                };
                return INTERCEPTORS.response.fail( error );
            }
        }
    );
}

METHODS.forEach(
    method => {
        MOCKS[ method ] = mockMethodFactory( method );
        FIXTURES[ method ] = {};
    }
);


export default {
    mocks: MOCKS,
    create: jest.fn().mockImplementation( () => {
        const instance  = {
            ...MOCKS,
            request( { url, method = 'get', data, params } ) {
                const args = [ url ];
                if ( undefined !== data ) {
                    args.push( data );
                }
                if ( undefined !== params ) {
                    args.push( params );
                }
                return instance[ method ].apply( null, args );
            },
            interceptors: {
                request: {
                    use( interceptor ) {
                        INTERCEPTORS.request = interceptor;
                    }
                },
                response: {
                    use( success, fail ) {
                        INTERCEPTORS.response = {
                            success,
                            fail
                        };
                    }
                }
            }
        };
        return instance;
    } ),
    use( method, url, data, status = 200, headers = {} ) {
        if ( !METHODS.includes( method ) ) {
            throw new Error( `Unknown method: "${method}"` );
        }
        FIXTURES[ method ][ url ] = {
            data,
            status,
            headers
        };
        return this;
    },

    useDefaultMocks() {
        return this;
    }
};
