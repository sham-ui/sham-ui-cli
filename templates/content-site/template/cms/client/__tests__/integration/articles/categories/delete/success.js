import setup, { app } from '../../../helpers';
import axios from 'axios';
jest.mock( 'axios' );

beforeEach( () => {
    jest.resetModules();
    jest.clearAllMocks();
} );

it( 'success delete cateogory', async() => {
    expect.assertions( 4 );

    const DI = setup();
    axios
        .useDefaultMocks()
        .use( 'get', '/categories', {
            categories: [
                { 'id': '1', 'name': 'Hello world!', 'slug': 'hello-world' },
                { 'id': '6', 'name': 'Название категории', 'slug': 'nazvanie-kategorii' }
            ]
        } );

    history.pushState( {}, '', 'http://client.example.com/articles/categories' );
    await app.start( DI );

    app.click( '[data-test-delete-button="1"]' );
    await app.waitRendering();

    app.checkMainPanel();

    axios
        .use( 'delete', '/categories/1' )
        .use( 'get', '/categories', {
            categories: [
                { 'id': '6', 'name': 'Название категории', 'slug': 'nazvanie-kategorii' }
            ]
        } );
    app.click( '[data-test-modal] [data-test-ok-button]' );
    await app.waitRendering();

    expect( axios.mocks.delete ).toHaveBeenCalledTimes( 1 );
    expect( axios.mocks.delete.mock.calls[ 0 ][ 0 ] ).toBe( '/categories/1' );
    app.checkMainPanel();
} );
