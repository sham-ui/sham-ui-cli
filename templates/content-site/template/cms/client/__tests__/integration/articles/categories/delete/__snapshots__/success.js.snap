// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`success delete cateogory 1`] = `
"<h4 class=\\"page-header\\"> Categories
  <!--0-->
</h4>
<div class=\\"page-body\\">
  <div class=\\"crud-page\\">
    <!--0-->
    <!--0-->
    <div class=\\"container\\">
      <button data-test-toggle-create-form=\\"\\" class=\\"mg-b-30\\">Show create new category form</button>
      <!--1-->
    </div>
    <div class=\\"container\\">
      <div class=\\"items-list\\">
        <div class=\\"panel\\">
          <table>
            <thead>
              <tr>
                <th>ID</th>
                <th>Name</th>
                <th>Slug</th>
                <th></th>
              </tr>
            </thead>
            <tbody>
              <tr class=\\"\\">
                <td class=\\"text-center\\">1</td>
                <td>Hello world!</td>
                <td>hello-world</td>
                <td class=\\"actions\\">
                  <button data-test-update-button=\\"1\\"> Edit </button>
                  <button class=\\"btn-danger\\" data-test-delete-button=\\"1\\"> Delete </button>
                </td>
              </tr>
              <tr class=\\"\\">
                <td class=\\"text-center\\">6</td>
                <td>Название категории</td>
                <td>nazvanie-kategorii</td>
                <td class=\\"actions\\">
                  <button data-test-update-button=\\"6\\"> Edit </button>
                  <button class=\\"btn-danger\\" data-test-delete-button=\\"6\\"> Delete </button>
                </td>
              </tr>
            </tbody>
          </table>
          <!--0-->
          <!--0-->
          <!--0-->
          <div class=\\"modal\\" data-test-modal=\\"\\"><span class=\\"close\\" title=\\"Close Modal\\" data-test-close-button=\\"\\">×</span>
            <!--0-->
            <div class=\\"modal-content\\">
              <div class=\\"container\\">
                <h4>Delete category</h4>
                <p class=\\"content\\">Are you sure you want to delete category \\"Hello world!\\"?</p>
                <div class=\\"footer\\">
                  <button data-test-ok-button=\\"\\" class=\\"btn-danger\\">Ok</button>
                  <button class=\\"btn-secondary\\" data-test-cancel-button=\\"\\">Cancel</button>
                </div>
              </div>
            </div>
          </div>
          <!--0-->
          <!--1-->
        </div>
      </div>
      <!--2-->
    </div>
  </div>
  <!--1-->
</div>
<!--0-->
<!--0-->
<!--0-->
<!--0-->"
`;

exports[`success delete cateogory 2`] = `
"<h4 class=\\"page-header\\"> Categories
  <!--0-->
</h4>
<div class=\\"page-body\\">
  <div class=\\"crud-page\\">
    <!--0-->
    <!--0-->
    <div class=\\"container\\">
      <button data-test-toggle-create-form=\\"\\" class=\\"mg-b-30\\">Show create new category form</button>
      <!--1-->
    </div>
    <div class=\\"container\\">
      <div class=\\"items-list\\">
        <div class=\\"panel\\">
          <table>
            <thead>
              <tr>
                <th>ID</th>
                <th>Name</th>
                <th>Slug</th>
                <th></th>
              </tr>
            </thead>
            <tbody>
              <tr class=\\"\\">
                <td class=\\"text-center\\">6</td>
                <td>Название категории</td>
                <td>nazvanie-kategorii</td>
                <td class=\\"actions\\">
                  <button data-test-update-button=\\"6\\"> Edit </button>
                  <button class=\\"btn-danger\\" data-test-delete-button=\\"6\\"> Delete </button>
                </td>
              </tr>
            </tbody>
          </table>
          <!--0-->
          <!--0-->
          <!--0-->
          <!--1-->
        </div>
      </div>
      <!--2-->
    </div>
  </div>
  <!--1-->
</div>
<!--0-->
<!--0-->
<!--0-->
<!--0-->"
`;
